### PR TITLE
sunxi: add NanoPi R1 support

### PR DESCRIPTION
**Pull request by jayanta525**
Specification:
-CPU: Allwinner H3, Quad-core Cortex-A7 Up to 1.2GHz
-DDR3 RAM: 512MB/1GB
-Network:
-   10/100/1000M Ethernet x 1，
-   10/100M Ethernet x 1
-WiFi: 802.11b/g/n, with SMA antenna interface
-Bluetooth：4.0 dual mode
-USB Host: Type-A x2
-MicroSD Slot x 1
-MicroUSB: for OTG and power input
-Debug Serial Port: 3Pin 2.54mm pitch pin-header
-UART: 4Pin 2.54mm pitch pin-header
-LED: LED x 3
-KEY: KEY x 1
-PC Size: 50.5 x 60mm
-Power Supply: DC 5V/2A

Installation:
-Use any burn tool to burn the disk image to SD Card.
-Insert SD Card and boot.

Signed-off-by: Jayantajit Gogoi <jayanta.gogoi525@gmail.com>

### CODE REVIEW

File: package/boot/uboot-sunxi/Makefile [hunk 1]
```diff
@@ -299,6 +305,7 @@ UBOOT_TARGETS := \
 	Linksprite_pcDuino3 \
 	Lamobo_R1 \
 	nanopi_m1_plus \
+  nanopi_r1 \
```
Comment (adrianschmutzler) [inline package/boot/uboot-sunxi/Makefile:308]: Indent.

File: package/boot/uboot-sunxi/patches/250-sun8i-h3-Nano-Pi-R1-add-device-tree.patch [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
++ *
++ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
++ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
++ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
++ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
++ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
++ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
++ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
++ *     OTHER DEALINGS IN THE SOFTWARE.
++ */
```
Comment (adrianschmutzler) [inline package/boot/uboot-sunxi/patches/250-sun8i-h3-Nano-Pi-R1-add-device-tree.patch:54]: Please add a SPDX license identifier instead.

File: package/boot/uboot-sunxi/patches/250-sun8i-h3-Nano-Pi-R1-add-device-tree.patch [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
++
```
Comment (adrianschmutzler) [inline package/boot/uboot-sunxi/patches/250-sun8i-h3-Nano-Pi-R1-add-device-tree.patch:55]: dts-v1 missing?

File: package/boot/uboot-sunxi/patches/250-sun8i-h3-Nano-Pi-R1-add-device-tree.patch [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
++#include "sun8i-h3-nanopi.dtsi"
++
++/ {
++	model = "FriendlyARM NanoPi R1";
++	compatible = "friendlyarm,nanopi-neo", "allwinner,sun8i-h3";
```
Comment (adrianschmutzler) [inline package/boot/uboot-sunxi/patches/250-sun8i-h3-Nano-Pi-R1-add-device-tree.patch:60]: compatible not matching model

File: package/boot/uboot-sunxi/patches/250-sun8i-h3-Nano-Pi-R1-add-device-tree.patch [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
++	};
++
++};
++
++&ehci0 {
++	status = "okay";
++};
++
++&pio {
++	gmac_power_pin_nanopi: gmac_power_pin@0 {
```
Comment (adrianschmutzler) [inline package/boot/uboot-sunxi/patches/250-sun8i-h3-Nano-Pi-R1-add-device-tree.patch:80]: Do we really need the device name in the label here?

File: target/linux/sunxi/base-files/etc/board.d/02_network [hunk 6]
```diff
@@ -18,6 +18,9 @@ case "$(board_name)" in
 "xunlong,orangepi-r1")
 	ucidef_set_interfaces_lan_wan "eth0" "eth1"
 	;;
+"friendlyarm,nanopi-r1")
+	ucidef_set_interfaces_lan_wan "eth1" "eth0"
+	;;
```
Comment (adrianschmutzler) [inline target/linux/sunxi/base-files/etc/board.d/02_network:23]: Please apply alphabetic sorting (I know it hasn't been before, so maybe just correct the file now.

File: target/linux/sunxi/image/cortex-a7.mk [hunk 7]
```diff
@@ -31,6 +31,16 @@ define Device/friendlyarm_nanopi-m1-plus
 endef
 TARGET_DEVICES += friendlyarm_nanopi-m1-plus
 
+define Device/sun8i-h3-nanopi-r1
```
Comment (adrianschmutzler) [inline target/linux/sunxi/image/cortex-a7.mk:34]: Please have a look at the other devices here. The name should be the compatible with "," replaced by "_", i.e. friendlyarm_nanopi-r1

File: target/linux/sunxi/image/cortex-a7.mk [hunk 8]
```diff
@@ -31,6 +31,16 @@ define Device/friendlyarm_nanopi-m1-plus
...
+  DEVICE_TITLE:=FriendlyArm NanoPi R1
```
Comment (adrianschmutzler) [inline target/linux/sunxi/image/cortex-a7.mk:35]: This needs to be replaced by the new syntax DEVICE_MODEL and DEVICE_VENDOR.

File: target/linux/sunxi/image/cortex-a7.mk [hunk 9]
```diff
@@ -31,6 +31,16 @@ define Device/friendlyarm_nanopi-m1-plus
...
+  DEVICE_PACKAGES:=kmod-rtc-sunxi kmod-usb2 kmod-usb-net kmod-usb-net-rtl8152 \
+	kmod-leds-gpio kmod-ledtrig-heartbeat \
```
Comment (adrianschmutzler) [inline target/linux/sunxi/image/cortex-a7.mk:37]: Please fill up lines until 80 chars, there is room for at least one additional package in this line.

File: target/linux/sunxi/image/cortex-a7.mk [hunk 10]
```diff
@@ -31,6 +31,16 @@ define Device/friendlyarm_nanopi-m1-plus
...
+	kmod-brcmfmac brcmfmac-firmware-43430-sdio wpad-basic
+  SUPPORTED_DEVICES:=friendlyarm,nanopi-r1
```
Comment (adrianschmutzler) [inline target/linux/sunxi/image/cortex-a7.mk:39]: This line can be dropped if you choose the proper name.

File: target/linux/sunxi/image/cortex-a7.mk [hunk 11]
```diff
@@ -31,6 +31,16 @@ define Device/friendlyarm_nanopi-m1-plus
...
+  SUNXI_DTS:=sun8i-h3-nanopi-r1
```
Comment (adrianschmutzler) [inline target/linux/sunxi/image/cortex-a7.mk:40]: Use SOC variable instead.

File: target/linux/sunxi/patches-5.4/062-ARM-dts-sun8i-add-support-for-Nano-Pi-R1.patch [hunk 12]
```diff
@@ -0,0 +1,210 @@
+--- a/arch/arm/boot/dts/Makefile
```
Comment (adrianschmutzler) [inline target/linux/sunxi/patches-5.4/062-ARM-dts-sun8i-add-support-for-Nano-Pi-R1.patch:1]: Same comment as above.

**Comment by adrianschmutzler** [comment]
Please add flashing instructions and remove the merge commit.

**Review by jayanta525 — COMMENTED** [review]
Made suggested changes

**Comment by adrianschmutzler** [comment]
Please merge all changes into a single proper commit, and remove those merge commits.

**Pull request closed without merging**